modules/tracking/samples/lasot_benchmark.py

```diff
@@ -0,0 +1,249 @@
+            # every video respectively
+            iou_list = []
+            pr_list = []
+            n_pr_list = []
+            frame_counter = len(video_sequence)
+
+            # For every frame in video
+            for number_of_the_frame, image in enumerate(video_sequence):
+                frame = cv.imread(os.path.join(
+                    args.path_to_dataset, video_name, "img", image))
+                for i in range(len(gt_bb)):
+                    gt_bb[i] = float(gt_bb[i])
+                gt_bb = tuple(gt_bb)
+
+                # Condition of tracker`s re-initialization
+                if ((number_of_the_frame + 1) % frames_for_reinit == 0) and (
+                        number_of_the_frame != 0):
+
+                    tracker, frames_for_reinit = init_tracker(tracker_name)
+                    init_once = False
+                    init_bb = gt_bb
+
+                if not init_once:
+                    init_state = tracker.init(frame, init_bb)
+                    init_once = True
+                init_state, new_bb = tracker.update(frame)
+
+                # Check for presence of object on the frame
+                # If no object on frame, we reduce number of
+                # accounted for evaluation frames
+                if gt_bb != (0, 0, 0, 0):
+                    # Calculation of coordinates of corners and centers
+                    # from [x, y, w, h] bounding boxes
+
+                    new_coords = list(calc_coords(new_bb).values())
+                    gt_coords = list(calc_coords(gt_bb).values())
+
+                    if args.visualization:
+                        cv.rectangle(frame, (int(new_coords[0]), int(
+                                     new_coords[2])), (int(new_coords[1]),
+                                     int(new_coords[3])), (200, 0, 0))
+                        cv.imshow("Tracking", frame)
+                        cv.waitKey(1)
+
+                    iou_list.append(get_iou(new_coords, gt_coords))
+                    pr_list.append(get_pr(new_coords, gt_coords))
+                    n_pr_list.append(get_norm_pr(
+                        new_coords, gt_coords, gt_bb[2], gt_bb[3]))
+                else:
+                    frame_counter -= 1
+
+                # Setting as ground truth bounding box from next frame
+                gt_bb = gt_file.readline().replace("\n", "").split(",")
+
+            # Calculating mean arithmetic value for specific video
+            iou_video += (np.fromiter([sum(i >= thr for i in iou_list).astype(
+                float) / frame_counter for thr in iou_thr], dtype=float))
+            pr_video += (np.fromiter([sum(i <= thr for i in pr_list).astype(
+                float) / frame_counter for thr in pr_thr], dtype=float))
+            n_pr_video += (np.fromiter([sum(i <= thr for i in n_pr_list).astype(
+                float) / frame_counter for thr in n_pr_thr], dtype=float))
```

Comment:
As for me we should compute it ones for all the ground truth BB and predicted ones. Metric should be invariant for number of videos. They could be merged in example into a single one.

Reply:
In original papers for that was used Pandas, but I rewrite Python generators without it. In this way we can skip frames with ground truth = (0, 0, 0, 0) more easily, so I think this way is better.

Reply:
I mean that there we need to compute it once for the outputs from videos but not for each video independently.

Reply:
Compute it separable for each video is better because like this we can see results for particular case if we need it, it is more flexible way to measure parameters of trackers.

Reply:
But is that the reliable final metric?

Reply:
I think yes. Calculations from the original code were reproduced very accurate.

Reply:
Ok, that's fine. Thanks for clarification!